### PR TITLE
RE SkyrimVM.h update events

### PR DESCRIPTION
**Pull request by Nightfallstorm**
`currentVMDaysPassed` is used for checking when to send update events for `queuedOnUpdateGameEvents`.

`currentVMMenuModeTime` is used to check when to send update events for `queuedOnUpdateEvents`

`updateTime` variable is calculated as follows:
RegisterFor(Single)Update -> (time argument in papyrus) * 1000, minimum 33 milliseconds
RegisterFor(Single)UpdateGameTime -> (time argument in papyrus) * 0.041666668 * 1000, minimum 6.9

### CODE REVIEW

File: include/RE/S/SkyrimVM.h [hunk 1]
```diff
@@ -139,6 +139,24 @@ namespace RE
...
+		public:
+			enum class UpdateType : bool
+			{
+				kRepeat = 0,					// RegisterForUpdate/RegisterForUpdateGameTime
+				kNoRepeat = 1,					// RegisterForSingleUpdate/RegisterForSingleUpdateGameTime
+			};
+			// members
+			std::uint32_t unk00;				// 00 - Possibly BSIntrusiveRefCounted?
+			UpdateType    updateType;           // 04
+			std::uint16_t pad_06;               // 06
```
Comment (powerof3) [inline include/RE/S/SkyrimVM.h]: underscore?

File: include/RE/S/SkyrimVM.h [hunk 2]
```diff
@@ -139,6 +139,24 @@ namespace RE
 	public:
 		inline static constexpr auto RTTI = RTTI_SkyrimVM;
 
+		struct UnkUpdateDataEvent
```
Comment (powerof3) [inline include/RE/S/SkyrimVM.h]: can get rid of the Unk bit

File: include/RE/S/SkyrimVM.h (hunk 1 above)
Comment (Nightfallstorm) [reply include/RE/S/SkyrimVM.h]: Updated

File: include/RE/S/SkyrimVM.h (hunk 2 above)
Comment (Nightfallstorm) [reply include/RE/S/SkyrimVM.h]: Updated :)

**Pull request merged**